frontend/src/components/QuerySentencePreview/QuerySentencePreview.tsx

```diff
@@ -63,6 +63,8 @@ const QuerySentencePreview: FunctionComponent<QuerySentencePreviewProps> = (prop
   const [previewOption, setPreviewOption] = useState<PreviewOption>('config');
   const [alert, setAlert] = useState<string[]>([]);
   const [validOptions, setValidOptions] = useState<AdvancedQueryOptions>();
+  const [resetStatus, setResetStatus] = useState<'clear-all' | 'clear-current'>('clear-current');
+  const [clearButtonText, setClearButtonText] = useState<string>(`Clear ${props.action}`);
```

Comment:
This is being used in a single place ... inline it. i.e.
`resetStatus === 'clear-all' ? 'Clear All' : 'Clear join/select/filter'`

Reply:
When done with that, deploy to staging for testing.